content/docs/extension-author-guide/create-extension/setup-local-environment.md

```diff
@@ -0,0 +1,75 @@
++++
+title="Set up your local environment"
+weight=401
++++
+
+<!-- test:suite=dockerfiles;weight=1 -->
+
+Let's walk through a build that uses extensions, step by step. We will see an image extension that installs `curl` on
+the builder image, and switches the run image to an image that has `curl` installed.
+
+### Ensure Docker is running
+
+<!-- test:exec -->
+```bash
+docker version
+```
+
+If you see output similar to the following, you're good to go! Otherwise, start Docker and check again.
+
+```
+Client: Docker Engine - Community
+ Version:           20.10.9
+ API version:       1.41
+ Go version:        go1.16.8
+ Git commit:        c2ea9bc
+ Built:             Mon Oct  4 16:08:29 2021
+ OS/Arch:           linux/amd64
+ Context:           default
+ Experimental:      true
+
+Server: Docker Engine - Community
+ Engine:
+  Version:          20.10.9
+  API version:      1.41 (minimum version 1.12)
+  Go version:       go1.16.8
+  Git commit:       79ea9d3
+  Built:            Mon Oct  4 16:06:34 2021
+  OS/Arch:          linux/amd64
+  Experimental:     false
+```
+
+### Setup workspace directory
+
+```bash
+cd <your preferred workspace directory>
+```
+
+### Ensure pack version supports image extensions
+
+<!-- test:exec -->
+```bash
+pack version
+```
+
+The version should be at least `0.28.0`
+
+### Enable experimental features in pack
+
+<!-- test:exec -->
+```bash
+pack config experimental true
+```
+
```

Comment:
Do we also need to ensure that `pack config lifecycle-image --unset` is run?  To ensure the user uses the latest lifecycle